contracts/Vault1.sol

```diff
@@ -1,6 +1,46 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
-contract Vault1 {
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "hardhat/console.sol";
+
+contract Vault1 is Ownable {
+    // what does this do exactly?
+    IERC20 public tokenContract;
+
+    constructor(address _contractAccount) {
+        tokenContract = IERC20(_contractAccount);
+    }
+
+    // should this be private?
+    mapping(address => uint256) public deposits;
     
+    function deposit(uint256 _amount) external hasEnoughTokens(_amount) isValidAmount(_amount) {
+        tokenContract.allowance(msg.sender, address(this));
```

Comment:
This allowance should be done directly by the user and should not be called from contract